llvm/lib/Support/regengine.inc

```diff
@@ -146,7 +146,9 @@ matcher(struct re_guts *g, const char *string, size_t nmatch,
 	const char *stop;
 
 	/* simplify the situation where possible */
-	if (g->cflags&REG_NOSUB)
+        if (!string)
```

Comment:
I don't think this code considers NULL string to be a valid input. I think this fix has to be in some caller higher up.

Reply:
I can move this to the function calling llvmregexec() within Regex.cpp ? Since the regex engine should be able to discard null strings as match-less for any regex.